### PR TITLE
Add TIL about finding controller#action from URI.

### PR DESCRIPTION
**Pull request by dleve123**
Explain how to find a controller#action from a URI string using `ActionDispatch::Routing::RouteSet#recognize_path`.

### CODE REVIEW

**Comment by lmatiolis** [comment]
Didn't know about it. Good TIL! :+1:

**Comment by tejasbubane** [comment]
:+1:

**Comment by JoelQ** [comment]
Thanks for sharing @dleve123 ! I wasn't aware of this method.

Thoughts on this @sgrif ?

**Comment by sgrif** [comment]
This is an internal API, and could break between versions without deprecation. I'm not sure why `RouteSet` is appearing on the API docs, it is not meant to be, as everything on it is private internal API.

File: rails/find_controller_action_from_uri.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+$ bundle exec rails console
+```
+Try out the following:
+```ruby
+Rails.application.routes.recognize_path('/posts') #=> { controller: 'posts', action: 'index' }
+```
+
+I discovered this method via a failed refactoring, so I honestly don't have a great use-case for `#recognize_path`.
+
+Nevertheless, I think `#recognize_path` is an interesting, undocumented, and [huge](https://github.com/rails/rails/blob/v4.1.8/actionpack/lib/action_dispatch/routing/route_set.rb#L681-L719) part `ActionDispatch::Routing::RouteSet`'s public API!
```
Comment (sgrif) [inline rails/find_controller_action_from_uri.md]: If something is undocumented, it is not part of the public API. `RouteSet` has no public API, as it is not part of the public API. I'm looking into why it's showing up on `api.rubyonrails.org` right now, but it's unintentional.

**Comment by sgrif** [comment]
https://github.com/rails/rails/commit/4a62724c59d387991055dbcb86d725e26a986fda

**Comment by dleve123** [comment]
@sgrif @JoelQ 

I have dug too far it seems - at least Rails' documentation has improved as a result!

I guess this repo doesn't want to endorse behavior that might change without notice - so should we close? Or is any knowledge applicable for TIL?

**Comment by sgrif** [comment]
@JoelQ Up to you? It's valid knowledge, but the Rails team would recommend against using it in any applications.

**Comment by JoelQ** [comment]
Thoughts on keeping the TIL but explicitly mentioning that it is part of the private API and shouldn't be used in applications?

**Comment by dleve123** [comment]
Also, more than happy to push up a commit that makes it clear that the method in question isn't part of any public API and shouldn't be used in production (but in the `rails console` for "poking around").

**Comment by dleve123** [comment]
@JoelQ woah - what timing! Will change the content accordingly.

**Comment by dleve123** [comment]
@JoelQ Updated. Do you want me to squash the commits?

File: rails/find_controller_action_from_uri.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Try out the following:
+```ruby
+Rails.application.routes.recognize_path('/posts') #=> { controller: 'posts', action: 'index' }
+```
+
+I discovered this method via a failed refactoring, so I honestly don't have a great use-case for `#recognize_path`.
+
+Nevertheless, I think `#recognize_path` is an interesting, undocumented, and [huge][4_1_8_recognize_path_source] part `ActionDispatch::Routing::RouteSet`'s API!
+
+[recognize_path_docs]: http://www.rubydoc.info/docs/rails/ActionDispatch/Routing/RouteSet:recognize_path
```
Comment (sgrif) [inline rails/find_controller_action_from_uri.md]: This link will 404 once we release 4.2.1

**Comment by JoelQ** [comment]
@dleve123 feel free to squash

**Comment by dleve123** [comment]
@JoelQ Soon to be dead link has been removed (it wasn't pointing to meaningful documentation anyways), and the entire TIL has been squashed into a single commit.

**Pull request merged**